flashinfer/gemm.py

```diff
@@ -1425,6 +1427,99 @@ def gemm_fp8_nt_groupwise(
     return out
 
 
+def trtllm_gemm_gen_module() -> JitSpec:
+    return gen_jit_spec(
+        "trtllm_gemm",
+        [
+            jit_env.FLASHINFER_CSRC_DIR / "trtllm_gemm_runner.cu",
+        ],
+        extra_cuda_cflags=[
+            "-DTLLM_GEN_EXPORT_INTERFACE",
+            "-DTLLM_ENABLE_CUDA",
+        ]
+        + sm100a_nvcc_flags,
+        extra_ldflags=["-lcuda"],
+    )
+
+
+@functools.cache
+def get_trtllm_gemm_module():
+    mod = trtllm_gemm_gen_module()
+    op = mod.build_and_load()
+    setup_cubin_loader(mod.get_library_path())
+    return op
+
+
+def gemm_fp8_blockwise_trtllm(
+    a: torch.Tensor,
+    b: torch.Tensor,
+    a_scale: torch.Tensor,
+    b_scale: torch.Tensor,
+    out: Optional[torch.Tensor] = None,
+) -> torch.Tensor:
+    r"""FP8 GEMM with block-wise scaling using TensorRT-LLM kernels.
```

Comment:
Can we also unify this to `gemm_fp8_nt_groupwise`? Seems functionality is similar.
https://github.com/flashinfer-ai/flashinfer/blob/b152d415cbba2cca8d3c7f176b618092ce2ef545/flashinfer/gemm.py#L1325-L1335

We can add another parameter `backend` which could be chosen between "cutlass" and "trtllm-gen".
For "trtllm-gen" the only supported "scale_granularity_mnk" is [128, 128, 128] at this moment.

Reply:
Done.